src/config.ts

```diff
@@ -294,3 +294,31 @@ export const saveConfiguration = async (
     );
   }
 };
+
+/**
+ * Checks if the default bedrock.yaml exists
+ *
+ * @param rootProjectPath Path to generate/update the the bedrock.yaml file in
+ */
+export const isBedrockFileValid = async (
+  rootProjectPath: string
+): Promise<[boolean, number | undefined]> => {
+  if (typeof rootProjectPath === "undefined" || rootProjectPath === "") {
+    throw new Error("Project root path is not valid");
+  }
+
+  const absProjectPath = path.resolve(rootProjectPath);
+
+  let bedrockFile: IBedrockFile | undefined;
+
+  try {
+    bedrockFile = await BedrockAsync(absProjectPath);
+  } catch (error) {
+    logger.error(error);
+    return [false, undefined];
+  }
+
+  logger.debug(`variableGroups length: ${bedrockFile?.variableGroups?.length}`);
+  logger.verbose(`bedrockFile: \n ${bedrockFile}`);
+  return [true, bedrockFile?.variableGroups?.length];
```

Comment:
you are overloading this function. I hope that you can return `true` or `false` here. and then do variableGroups length check in the caller function. When we overload function, you can see that you needed to return things like undefined. thanks.

Reply:
It is a deep check i.e. not only a file exist, but need to make sure variable groups exist in the file. In this scenario, the caller does't need the file contents and true/false is sufficient. Perhaps we can change the second one to `boolean` and remove the `undefined`.  Let me know your thoughts on this.

Reply:
I am misled by the function name, isBedrockFileValid. As the function name suggest, I thought that we are checking if bedrock file exist or not :-) I think we can do this.
```
interface IBedrockFileStat {
  exist: boolean;
  hasVariableGroups: boolean;
}

export const bedrockFileStat = async (
  rootProjectPath: string
): Promise<boolean> => {
  if (!rootProjectPath) {
    throw new Error("Project root path is not valid");
  }

  const absProjectPath = path.resolve(rootProjectPath);

  let bedrockFile: IBedrockFile | undefined;

  try {
    const bedrockFile = await BedrockAsync(absProjectPath);
    return {
      exist: true,
      hasVariableGroups: bedrockFile?.variableGroups?.length
    }
  } catch (error) {
    logger.error(error);
    return {
      exist: false,
    }
  }
}
```

this enables other team members to reuse and enhance this function get more stats from bedrock file.
and we do not return array, we return a well defined interface.

what do you think?

Reply:
Yes, the name was misleading. I thought about it and am going to refactor it to check whether dependency commands were executed.  One of the goal it to centralize the error messages too so they won't be repeated.

Reply:
Discussed with @mtarng and the decided to keep it as `bedrock file info` and use it to validate the project commands order.

Reply:
sorry, I do not understand `keep it as bedrock file info`. Do you mind elaborate?

Reply:
I was thinking to implement command sequence validator using this logic, but decided to keep the method name as bedrock file that may be used for other checks in the future.

Reply:
ok